vllm/v1/core/sched/scheduler.py

```diff
@@ -76,9 +76,12 @@ def __init__(
         # Priority queues for requests.
         self.waiting: deque[Request] = deque()
         self.running: list[Request] = []
-        # The requests that have been scheduled and are being executed
-        # by the executor.
-        self.scheduled_req_ids: set[str] = set()
+
+        # req_id -> Number of times the request has been scheduled.
+        # With PP, when the input prompt is divided into chunks, we can
+        # schedule a new chunk even before the previous chunk has completed
+        # the full pipeline stages. This helps reduce TTFT.
+        self.scheduled_req_ids: dict[str, int] = defaultdict(int)
```

Comment:
Dumb question: What is `scheduled_req_ids` used for? IIUC, this PR eliminates its usage.

Reply:
This may be a good catch. Since we now simply check whether num_new_tokens=0 to determine whether we could continue scheduling a request, we may not need this anymore...

Reply:
@comaniac Then, can you please remove it?